Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -54,6 +54,22 @@
 #include <wtf/URL.h>
 #include <wtf/text/AtomString.h>
 #include <wtf/text/AtomStringHash.h>
+#include <wtf/text/StringToIntegerConversion.h>
+
+// We shouldn't accept media that the player can't actually play.
+// AAC supports up to 96 channels.
+#define MEDIA_MAX_AAC_CHANNELS 96
+#if PLATFORM(RPI)
+// Raspberry Pi only supports up to 1080p@30fps hardware video decoding.
+#define MEDIA_MAX_WIDTH 1920.0f
+#define MEDIA_MAX_HEIGHT 1080.0f
+#define MEDIA_MAX_FRAMERATE 30.0f
```

Comment:
There is nothing specific about the RaspberryPi here. The option can be useful for other platforms where the available hardware decoders (or the software based decoding) are not able to keep up with higher video resolutions.

I would prefer to call this option something like `USE(FULLHD_VIDEO_DECODING_LIMIT)`&mdash;if anybody else has a better name which transmits the same idea, that would be fine as well.

Reply:
I agree with you, it should be possible to use this settings also on other platforms (which support only limited resolutions), so as conclusion: I should change the name of the option to USE(FULLHD_VIDEO_DECODING_LIMIT) instead of PLATFORM(RPI).
The question is how it should be activated? Should it be set on upper level (buildroot, yocto config)?
Or in case RPi, detect it on cmake level and set for RPi (but what about other platform?)?

I think that detection of the supported playback resolution on the configuration level (cmake) can be not trivial task.

Reply:
What I meant with autodetection is to not have a CMake option, but make the C++ code detect the capabilities of the decoder and adapt at runtime without needing to rebuild.

At any rate, the CMake option must be specified explicitly by the build system or the person doing the build. We do *not* want to detect things at the CMake level: it would be very error prone, specially when cross-compilation is involved.

Reply:
Buildroot can perfectly export this new general limit flag, and can be added to the the relevant device defconfigs. Would also put FHD as suffix, so that e.g. 4K can be added later.